File: Greedy/Java/Assign Cookies.java
// Problem link: https://leetcode.com/problems/assign-cookies/description/
// Time: O(nlogn)
// Space: O(n) for sorting
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int i = 0, j = 0;

        while(i < g.length && j < s.length){
            if(g[i] <= s[j]){
                i++;
            }
            j++;
        }

        return i;
    }
}

File: Greedy/Python/Assign Cookies.py
# Problem link: https://leetcode.com/problems/assign-cookies/description/
# Time: O(nlogn)
# Space: O(n) for sorting
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()
        i = j = 0

        while i < len(g) and j < len(s):
            if s[j] >= g[i]:
                i += 1
            j += 1
        
        return i
